core/src/color.rs

```diff
@@ -33,11 +36,21 @@ impl Color {
         a: 0.0,
     };
 
+    /// New Color with range checks
+    pub fn new(r: f32, g: f32, b: f32, a: f32) -> Color {
+        Color {
+            r: clamp(r),
+            g: clamp(g),
+            b: clamp(b),
+            a: clamp(a),
+        }
```

Comment:
Maybe debug assertions are a better idea? I think clamping here normally indicates a logic error from the user of the library.